### PR TITLE
T17886

### PR DESCRIPTION
**Pull request by cosimoc**
https://phabricator.endlessm.com/T17886

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -1300,14 +1300,8 @@ function appendArtworkCardsFromShardsAndItems(shards, items, proxy, type, direct
                         title: entry.title,
                         author: entry.author,
                         thumbnail: thumbnail,
-                        desktop_id: proxy.desktopId,
-                        bus_name: proxy.busName,
-                        knowledge_search_object_path: proxy.knowledgeSearchObjectPath,
-                        knowledge_app_id: proxy.knowledgeAppId,
-                        uri: entry.ekn_id,
```
Comment (smspillaz) [inline src/main.js]: Hmm, this is a good point (the perils of me copy-pasting code...). That being said, I wonder if artwork cards should "go" somewhere when clicked, to the WikiArt app, perhaps? In that case, `DiscoveryFeedKnowledgeArtworkCardStore` should be a subclass of `DiscoveryFeedKnowledgeAppCardStore` ?

Comment (cosimoc) [inline src/main.js]: It absolutely should eventually; feel free to open a follow up story or take this up and change to make it so.

File: src/main.js [hunk 2]
```diff
@@ -893,7 +893,7 @@ const DiscoveryFeedKnowledgeAppCard = new Lang.Class({
             loadKnowledgeAppContent(this._app,
                                     this._knowledgeSearchProxy,
                                     this.model.uri,
-                                    'knowledge_content');
+				    this.contentType);
```
Comment (smspillaz) [inline src/main.js]: Looks like this is using tabs

File: src/main.js [hunk 3]
```diff
@@ -921,26 +925,21 @@ const DiscoveryFeedKnowledgeArtworkCard = new Lang.Class({
...
         this.get_style_context().add_class('artwork');
+    },
+
+    createDescription: function() {
+        return new DiscoveryFeedAppContentDescription({
+            title: this.model.title,
+            synopsis: 'by ' + this.model.author,
+            app_name: 'Masterpiece of the Day'.toUpperCase()
+        });
+    },
```
Comment (smspillaz) [inline src/main.js]: Not going to block on this, but my general preference here is to prefer injecting things like this into the base class as opposed to overriding.

File: src/stores.js [hunk 4]
```diff
@@ -237,6 +199,25 @@ const DiscoveryFeedKnowledgeAppCardStore = new Lang.Class({
...
+    Name: 'DiscoveryFeedKnowlegeArtworkCardStore',
+    Extends: DiscoveryFeedKnowledgeAppCardStore,
+    Properties: {
+        author: GObject.ParamSpec.string('author',
+                                           '',
+                                           '',
+                                           GObject.ParamFlags.READWRITE |
+                                           GObject.ParamFlags.CONSTRUCT_ONLY,
+                                           '')
+    },
```
Comment (smspillaz) [inline src/stores.js]: Seems misaligned

**Pull request merged**